porch/apiserver/pkg/registry/porch/background.go

```diff
@@ -84,6 +84,8 @@ loop:
 			if !eventOk {
 				klog.Errorf("Watch event stream closed. Will restart watch from bookmark %q", bookmark)
 				watcher.Stop()
+				// Avoid busy-spinning
+				time.Sleep(1 * time.Second)
```

Comment:
(perhaps to start we can put the re-connect logic in the background ticker handler. it would happen every minute but if we are having trouble getting the watcher, things are likely ripe for exponential backoff or long-ish timeouts anyway)